### ISSUE TITLE
Failed to Convert YOLOv8n Model to TFLite Using CLI and Sample Code

### ISSUE THREAD

**Issue by henry3556108** [ultralytics/ultralytics#14619]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

I attempted to convert the YOLOv8n model to TFLite format using both the CLI and sample code provided. However, the conversion process failed on two different systems. Below are the details of my attempts:

### Windows Error Log
```
TensorFlow SavedModel: starting TFLite export with onnx2tf 1.20.0...
ERROR: The trace log is below.
Traceback (most recent call last):
File "C:\Users\hti07015\AppData\Local\anaconda3\envs\yolo_v8\lib\site-packages\onnx2tf\utils\common_functions.py", line 310, in print_wrapper_func
result = func(*args, **kwargs)
File "C:\Users\hti07015\AppData\Local\anaconda3\envs\yolo_v8\lib\site-packages\onnx2tf\utils\common_functions.py", line 383, in inverted_operation_enable_disable_wrapper_func
result = func(*args, **kwargs)
File "C:\Users\hti07015\AppData\Local\anaconda3\envs\yolo_v8\lib\site-packages\onnx2tf\utils\common_functions.py", line 53, in get_replacement_parameter_wrapper_func
func(*args, **kwargs)
File "C:\Users\hti07015\AppData\Local\anaconda3\envs\yolo_v8\lib\site-packages\onnx2tf\ops\Resize.py", line 417, in make_node
resized_tensor = Lambda(
File "C:\Users\hti07015\AppData\Local\anaconda3\envs\yolo_v8\lib\site-packages\tensorflow\python\keras\engine\base_layer.py", line 1045, in call
outputs = call_fn(inputs, *args, **kwargs)
File "C:\Users\hti07015\AppData\Local\anaconda3\envs\yolo_v8\lib\site-packages\tensorflow\python\keras\layers\core.py", line 913, in call
result = self.function(inputs, **kwargs)
File "C:\Users\hti07015\AppData\Local\anaconda3\envs\yolo_v8\lib\site-packages\onnx2tf\utils\common_functions.py", line 1142, in upsampling2d_nearest
return tf.compat.v1.image.resize_nearest_neighbor(
File "C:\Users\hti07015\AppData\Local\anaconda3\envs\yolo_v8\lib\site-packages\tensorflow\python\ops\image_ops_impl.py", line 4769, in resize_nearest_neighbor
return gen_image_ops.resize_nearest_neighbor(
File "C:\Users\hti07015\AppData\Local\anaconda3\envs\yolo_v8\lib\site-packages\tensorflow\python\ops\gen_image_ops.py", line 4376, in resize_nearest_neighbor
return resize_nearest_neighbor_eager_fallback(
File "C:\Users\hti07015\AppData\Local\anaconda3\envs\yolo_v8\lib\site-packages\tensorflow\python\ops\gen_image_ops.py", line 4414, in resize_nearest_neighbor_eager_fallback
_attr_T, (images,) = _execute.args_to_matching_eager([images], ctx, [_dtypes.int8, _dtypes.uint8, _dtypes.int16, _dtypes.uint16, _dtypes.int32, _dtypes.int64, _dtypes.half, _dtypes.float32, _dtypes.float64, _dtypes.bfloat16, ])
File "C:\Users\hti07015\AppData\Local\anaconda3\envs\yolo_v8\lib\site-packages\tensorflow\python\eager\execute.py", line 251, in args_to_matching_eager
tensor = tensor_conversion_registry.convert(t)
File "C:\Users\hti07015\AppData\Local\anaconda3\envs\yolo_v8\lib\site-packages\tensorflow\python\framework\tensor_conversion_registry.py", line 234, in convert
ret = conversion_func(value, dtype=dtype, name=name, as_ref=as_ref)
File "C:\Users\hti07015\AppData\Local\anaconda3\envs\yolo_v8\lib\site-packages\tensorflow\python\framework\constant_op.py", line 324, in _constant_tensor_conversion_function
return constant(v, dtype=dtype, name=name)
File "C:\Users\hti07015\AppData\Local\anaconda3\envs\yolo_v8\lib\site-packages\tensorflow\python\framework\constant_op.py", line 263, in constant
return _constant_impl(value, dtype, shape, name, verify_shape=False,
File "C:\Users\hti07015\AppData\Local\anaconda3\envs\yolo_v8\lib\site-packages\tensorflow\python\framework\constant_op.py", line 275, in _constant_impl
return _constant_eager_impl(ctx, value, dtype, shape, verify_shape)
File "C:\Users\hti07015\AppData\Local\anaconda3\envs\yolo_v8\lib\site-packages\tensorflow\python\framework\constant_op.py", line 285, in _constant_eager_impl
t = convert_to_eager_tensor(value, ctx, dtype)
File "C:\Users\hti07015\AppData\Local\anaconda3\envs\yolo_v8\lib\site-packages\tensorflow\python\framework\constant_op.py", line 98, in convert_to_eager_tensor
return ops.EagerTensor(value, ctx.device_name, dtype)
File "C:\Users\hti07015\AppData\Local\anaconda3\envs\yolo_v8\lib\site-packages\keras\src\engine\keras_tensor.py", line 285, in array
raise TypeError(
TypeError: You are passing KerasTensor(type_spec=TensorSpec(shape=(1, 20, 20, 512), dtype=tf.float32, name=None), name='tf.math.multiply_119/Mul:0', description="created by layer 'tf.math.multiply_119'"), an intermediate Keras symbolic input/output, to a TF API that does not allow registering custom dispatchers, such as tf.cond, tf.function, gradient tapes, or tf.map_fn. Keras Functional model construction only supports TF API calls that do support dispatching, such as tf.math.add or tf.reshape. Other APIs cannot be called directly on symbolic Kerasinputs/outputs. You can work around this limitation by putting the operation in a custom Keras layer call and calling that layer on this symbolic input/output.

ERROR: input_onnx_file_path: yolov8s.onnx
ERROR: onnx_op_name: /model.10/Resize
ERROR: Read this and deal with it. https://github.com/PINTO0309/onnx2tf#parameter-replacement
ERROR: Alternatively, if the input OP has a dynamic dimension, use the -b or -ois option to rewrite it to a static shape and try again.
ERROR: If the input OP of ONNX before conversion is NHWC or an irregular channel arrangement other than NCHW, use the -kt or -kat option.
ERROR: Also, for models that include NonMaxSuppression in the post-processing, try the -onwdt option.
```

### Windows package version
```
keras2.13.1
networkx3.1
numpy1.24.3
onnx1.16.1
onnx-graphsurgeon0.5.2
onnx2tf1.20.0
onnxruntime1.18.1
onnxslim0.1.32
opencv-python4.10.0.84
openvino2024.2.0
openvino-telemetry2024.1.0
pandas2.0.3
pillow10.4.0
sng4onnx1.0.4
tensorboard2.13.0
tensorboard-data-server0.7.2
tensorflow2.13.0
tensorflow-estimator2.13.0
tensorflow-hub0.12.0
tensorflow-intel2.13.0
tensorflow-io-gcs-filesystem0.31.0
tensorflowjs3.18.0
termcolor2.4.0
tf-keras2.15.0
tflite-support0.4.3
torch2.1.2
torchvision0.16.2
ultralytics8.2.58
ultralytics-thop==2.0.0
```

### Mac Error Log

```
TensorFlow SavedModel: starting TFLite export with onnx2tf 1.20.0...
ERROR: The trace log is below.
Traceback (most recent call last):
  File "/Users/hongyicheng/.pyenv/versions/3.8.10/lib/python3.8/site-packages/onnx2tf/utils/common_functions.py", line 310, in print_wrapper_func
    result = func(*args, **kwargs)
  File "/Users/hongyicheng/.pyenv/versions/3.8.10/lib/python3.8/site-packages/onnx2tf/utils/common_functions.py", line 383, in inverted_operation_enable_disable_wrapper_func
    result = func(*args, **kwargs)
  File "/Users/hongyicheng/.pyenv/versions/3.8.10/lib/python3.8/site-packages/onnx2tf/utils/common_functions.py", line 53, in get_replacement_parameter_wrapper_func
    func(*args, **kwargs)
  File "/Users/hongyicheng/.pyenv/versions/3.8.10/lib/python3.8/site-packages/onnx2tf/ops/Resize.py", line 417, in make_node
    resized_tensor = Lambda(
  File "/Users/hongyicheng/.pyenv/versions/3.8.10/lib/python3.8/site-packages/tensorflow/python/keras/engine/base_layer.py", line 1045, in __call__
    outputs = call_fn(inputs, *args, **kwargs)
  File "/Users/hongyicheng/.pyenv/versions/3.8.10/lib/python3.8/site-packages/tensorflow/python/keras/layers/core.py", line 913, in call
    result = self.function(inputs, **kwargs)
  File "/Users/hongyicheng/.pyenv/versions/3.8.10/lib/python3.8/site-packages/onnx2tf/utils/common_functions.py", line 1142, in upsampling2d_nearest
    return tf.compat.v1.image.resize_nearest_neighbor(
  File "/Users/hongyicheng/.pyenv/versions/3.8.10/lib/python3.8/site-packages/tensorflow/python/ops/image_ops_impl.py", line 4769, in resize_nearest_neighbor
    return gen_image_ops.resize_nearest_neighbor(
  File "/Users/hongyicheng/.pyenv/versions/3.8.10/lib/python3.8/site-packages/tensorflow/python/ops/gen_image_ops.py", line 3858, in resize_nearest_neighbor
    return resize_nearest_neighbor_eager_fallback(
  File "/Users/hongyicheng/.pyenv/versions/3.8.10/lib/python3.8/site-packages/tensorflow/python/ops/gen_image_ops.py", line 3896, in resize_nearest_neighbor_eager_fallback
    _attr_T, (images,) = _execute.args_to_matching_eager([images], ctx, [_dtypes.int8, _dtypes.uint8, _dtypes.int16, _dtypes.uint16, _dtypes.int32, _dtypes.int64, _dtypes.half, _dtypes.float32, _dtypes.float64, _dtypes.bfloat16, ])
  File "/Users/hongyicheng/.pyenv/versions/3.8.10/lib/python3.8/site-packages/tensorflow/python/eager/execute.py", line 251, in args_to_matching_eager
    tensor = tensor_conversion_registry.convert(t)
  File "/Users/hongyicheng/.pyenv/versions/3.8.10/lib/python3.8/site-packages/tensorflow/python/framework/tensor_conversion_registry.py", line 234, in convert
    ret = conversion_func(value, dtype=dtype, name=name, as_ref=as_ref)
  File "/Users/hongyicheng/.pyenv/versions/3.8.10/lib/python3.8/site-packages/tensorflow/python/framework/constant_op.py", line 324, in _constant_tensor_conversion_function
    return constant(v, dtype=dtype, name=name)
  File "/Users/hongyicheng/.pyenv/versions/3.8.10/lib/python3.8/site-packages/tensorflow/python/framework/constant_op.py", line 263, in constant
    return _constant_impl(value, dtype, shape, name, verify_shape=False,
  File "/Users/hongyicheng/.pyenv/versions/3.8.10/lib/python3.8/site-packages/tensorflow/python/framework/constant_op.py", line 275, in _constant_impl
    return _constant_eager_impl(ctx, value, dtype, shape, verify_shape)
  File "/Users/hongyicheng/.pyenv/versions/3.8.10/lib/python3.8/site-packages/tensorflow/python/framework/constant_op.py", line 285, in _constant_eager_impl
    t = convert_to_eager_tensor(value, ctx, dtype)
  File "/Users/hongyicheng/.pyenv/versions/3.8.10/lib/python3.8/site-packages/tensorflow/python/framework/constant_op.py", line 98, in convert_to_eager_tensor
    return ops.EagerTensor(value, ctx.device_name, dtype)
  File "/Users/hongyicheng/.pyenv/versions/3.8.10/lib/python3.8/site-packages/keras/src/engine/keras_tensor.py", line 285, in __array__
    raise TypeError(
TypeError: You are passing KerasTensor(type_spec=TensorSpec(shape=(1, 20, 20, 256), dtype=tf.float32, name=None), name='tf.math.multiply_119/Mul:0', description="created by layer 'tf.math.multiply_119'"), an intermediate Keras symbolic input/output, to a TF API that does not allow registering custom dispatchers, such as `tf.cond`, `tf.function`, gradient tapes, or `tf.map_fn`. Keras Functional model construction only supports TF API calls that *do* support dispatching, such as `tf.math.add` or `tf.reshape`. Other APIs cannot be called directly on symbolic Kerasinputs/outputs. You can work around this limitation by putting the operation in a custom Keras layer `call` and calling that layer on this symbolic input/output.

ERROR: input_onnx_file_path: yolov8n.onnx
ERROR: onnx_op_name: /model.10/Resize
ERROR: Read this and deal with it. https://github.com/PINTO0309/onnx2tf#parameter-replacement
ERROR: Alternatively, if the input OP has a dynamic dimension, use the -b or -ois option to rewrite it to a static shape and try again.
ERROR: If the input OP of ONNX before conversion is NHWC or an irregular channel arrangement other than NCHW, use the -kt or -kat option.
ERROR: Also, for models that include NonMaxSuppression in the post-processing, try the -onwdt option.
```

### Mac python pacakge version
```
keras==2.13.1
numpy==1.24.3
onnx==1.16.1
onnx-graphsurgeon==0.5.2
onnx2tf==1.20.0
onnxruntime==1.18.1
onnxslim==0.1.32
opencv-python==4.10.0.84
opencv-python-headless==4.10.0.84
opt-einsum==3.3.0
tensorboard==2.13.0
tensorboard-data-server==0.7.2
tensorflow==2.13.0
tensorflow-estimator==2.13.0
tensorflow-macos==2.13.0
termcolor==2.4.0
tf-keras==2.15.0
tflite-support==0.1.0a1
tifffile==2023.7.10
torch==2.3.1
torchvision==0.18.1
ultralytics==8.2.62
ultralytics-thop==2.0.0
```

### Environment

Macbook M1 Pro with Ventura 13.6 and intel i5-14500 with Windows OS

### Minimal Reproducible Example

Steps to Reproduce:

Used the CLI command 
    `yolo export model=yolov8n.pt format=tflite`
Sample code 
```python
from ultralytics import YOLO

# Load the YOLOv8 model
model = YOLO("yolov8n.pt")

# Export the model to TFLite format
model.export(format="tflite")  # creates 'yolov8n_float32.tflite'

# Load the exported TFLite model
tflite_model = YOLO("yolov8n_float32.tflite")
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by ambitious-octopus** [comment]
Hey @henry3556108, can you please share the output of: `yolo checks`? I just tested with the latest version and everything works fine!

**Comment by glenn-jocher** [comment]
@ambitious-octopus please add non-reproducible tags to bug reports that we can not reproduce locally.

**Comment by henry3556108** [comment]
@glenn-jocher Sorry, I fixed it by switch python's version to 3.9.
It seems that python 3.9 or higher is required to convert to tflite model

**Comment by glenn-jocher** [comment]
@henry3556108 thank you for the update. Yes, Python 3.9 or higher is required for converting to TFLite models. If you encounter any further issues, please ensure you are using the latest versions of all relevant packages. If the problem persists, feel free to reach out.

**Comment by yll1** [comment]
> @glenn-jocher Sorry, I fixed it by switch python's version to 3.9. It seems that python 3.9 or higher is required to convert to tflite model

@henry3556108 
As you said, when you used python3.9, you successfully converted the model to tftile modle mode, can you post the version numbers of tensorflow, onnx2tf and ultralytics at this time or can you post your pip list at this time? Thank you very much

**Comment by glenn-jocher** [comment]
It's great to hear you resolved the issue with Python 3.9. For version details, please refer to the latest documentation or your current environment setup. If you need further assistance, feel free to ask!
